spring-websockets/src/test/java/com/baeldung/debugwebsockets/WebsocketIntegrationTest.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.web.socket.messaging.WebSocketStompClient;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * This should be part of integration test suite.
+ * The test starts the server and then connects to the Websocket. Then verifies if the messages are received from the Websocket.
+ * This test is inspired from: https://github.com/spring-guides/gs-messaging-stomp-websocket/blob/main/complete/src/test/java/com/example/messagingstompwebsocket/GreetingIntegrationTests.java
+ */
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class WebsocketIntegrationTest {
+
+    WebSocketClient client;
+    WebSocketStompClient stompClient;
+    @LocalServerPort
+    private int port;
+    private static final Logger logger= LoggerFactory.getLogger(WebsocketIntegrationTest.class);
+
+    @BeforeEach
+    public void setup() {
+        logger.info("Setting up the tests ...");
+        client = new StandardWebSocketClient();
+        stompClient = new WebSocketStompClient(client);
+        stompClient.setMessageConverter(new MappingJackson2MessageConverter());
+    }
+
+    @Test
+    void givenWebsocket_whenMessage_thenVerifyMessage() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<Throwable> failure = new AtomicReference<>();
+        StompSessionHandler sessionHandler = new StompSessionHandler() {
+
+            @Override
+            public Type getPayloadType(StompHeaders headers) {
+                return null;
+            }
+
+            @Override
+            public void handleFrame(StompHeaders headers, Object payload) {
+
+            }
+
+            @Override
+            public void afterConnected(StompSession session, StompHeaders connectedHeaders) {
+                logger.info("Connected to the Websocket ...");
+                session.subscribe("/topic/ticks", new StompFrameHandler() {
+                    @Override
+                    public Type getPayloadType(StompHeaders headers) {
+                        return Map.class;
+                    }
+
+                    @Override
+                    public void handleFrame(StompHeaders headers, Object payload) {
+                        try {
+
+                            assertThat(payload).isNotNull();
+                            assertThat(payload).isInstanceOf(Map.class);
+
+                            @SuppressWarnings("unchecked")
+                            Map<String, Integer> map = (Map<String, Integer>) payload;
+
+                            assertThat(map).containsKey("HPE");
+                            assertThat(map.get("HPE")).isInstanceOf(Integer.class);
+
```

Comment:
Remove empty lines at beginning and end of block

Reply:
Thank You @KevinGilmore.  Made the change